test/AbstractMemoryTest.cpp

```diff
@@ -0,0 +1,209 @@
+  ad_utility::Memory m1;
+  checkAllMemoryGetter(m1, MemorySize{0uL, 0.0, 0.0, 0.0, 0.0, 0.0});
+
+  // Non-default constructor.
+  ad_utility::Memory m2(1024);
+  checkAllMemoryGetter(m2,
+                       MemorySize{1024uL, 1.0, 0.0009765625, 9.5367431640625e-7,
+                                  9.31322574615478515625e-10,
+                                  9.094947017729282379150390625e-13});
+}
+
+TEST(AbstractMemory, SizeTAssignmentOperator) {
+  ad_utility::Memory m;
+  checkAllMemoryGetter(m, MemorySize{0uL, 0.0, 0.0, 0.0, 0.0, 0.0});
+
+  m = 1_Byte;
+  checkAllMemoryGetter(m, MemorySize{1uL, 0.0009765625, 9.5367431640625e-7,
+                                     9.31322574615478515625e-10,
+                                     9.094947017729282379150390625e-13,
+                                     8.8817841970012523233890533447265625e-16});
+
+  m = 1_KB;
+  checkAllMemoryGetter(m,
+                       MemorySize{1024uL, 1.0, 0.0009765625, 9.5367431640625e-7,
+                                  9.31322574615478515625e-10,
+                                  9.094947017729282379150390625e-13});
+
+  m = 1_MB;
+  checkAllMemoryGetter(
+      m, MemorySize{1048576uL, 1024, 1.0, 0.0009765625, 9.5367431640625e-7,
+                    9.31322574615478515625e-10});
+
+  m = 1_GB;
+  checkAllMemoryGetter(m, MemorySize{1073741824uL, 1048576.0, 1024.0, 1.0,
+                                     0.0009765625, 9.5367431640625e-7});
+
+  m = 1_TB;
+  checkAllMemoryGetter(m, MemorySize{1099511627776uL, 1073741824.0, 1048576.0,
+                                     1024.0, 1.0, 0.0009765625});
+
+  m = 1_PB;
+  checkAllMemoryGetter(m, MemorySize{1125899906842624uL, 1099511627776.0,
+                                     1073741824.0, 1048576.0, 1024.0, 1.0});
+}
+
+// For tests, where one is converted into the other and vice-versa.
+struct MemoryAmountAndStringRepresentation {
+  size_t memoryAmount_;
+  std::string stringRepresentation;
+};
+
+static std::vector<MemoryAmountAndStringRepresentation>
+generalAsStringTestCases() {
+  return {{50_Byte, "50 Byte"}, {2_KB, "2 KB"},     {1.5_KB, "1.5 KB"},
+          {2_MB, "2 MB"},       {1.5_MB, "1.5 MB"}, {2_GB, "2 GB"},
+          {1.5_GB, "1.5 GB"},   {2_TB, "2 TB"},     {1.5_TB, "1.5 TB"},
+          {2_PB, "2 PB"},       {1.5_PB, "1.5 PB"}};
+}
+
+TEST(AbstractMemory, AsString) {
+  // Creates an instance with given amount of memory noted and checks the
+  // expected string representation.
+  auto doTest = [](const MemoryAmountAndStringRepresentation& testCase) {
+    ASSERT_STREQ(ad_utility::Memory(testCase.memoryAmount_).asString().c_str(),
+                 testCase.stringRepresentation.data());
+  };
+
+  std::ranges::for_each(generalAsStringTestCases(), doTest);
+
+  // Check, if it always uses the biggest unit.
+  doTest({4096_Byte, "4 KB"});
+  doTest({4096_KB, "4 MB"});
+  doTest({4096_MB, "4 GB"});
+  doTest({4096_GB, "4 TB"});
+  doTest({4096_TB, "4 PB"});
+}
```

Comment:
I think we don't need the space between the number and the unit, so rather `4 KB`

Reply:
So, you rather want to tests to use, for example, `4KB` instead of `4 KB`?

Reply:
Yes, 
but let me talk to @hannahbast later, whether we want `4kB`, `4KB`, `4 kB` or `4 KB` or something entirely else.
Same goes for the operator literals so `4_KB` vs `4_kB`.

Reply:
Alright, can do. However, I have to point out that, currently, we support all those representations. Our grammar is case insensitive and ignores whitespaces.

Reply:
Is like [this]([9ed1da8](https://github.com/ad-freiburg/qlever/pull/1032/commits/9ed1da825f6312e2089e9402e2be442696bfa63d) okay?

Reply:
The format is great now.